testng/src/main/java/cucumber/api/testng/TestNGCucumberRunner.java

```diff
@@ -68,6 +69,10 @@ public void runCucumber(CucumberFeature cucumberFeature) {
         }
     }
 
+    public void runScenario(PickleEvent pickle) throws Throwable {
+        runtime.runPickle(pickle);
```

Comment:
I've added a test that verifies the working of the new listener when a scenario contains undefined steps and the 'strict' flag is set.